test/cmdlineTests.sh

```diff
@@ -49,6 +49,30 @@ do
     test -z "$output" -a "$failed" -eq 0
 done
 
+echo "Compiling all examples from the documentation..."
+TMPDIR=$(mktemp -d)
+(
+    set -e
+    cd "$REPO_ROOT"
+    REPO_ROOT=$(pwd) # make it absolute
+    cd "$TMPDIR"
+    "$REPO_ROOT"/scripts/isolate_tests.py "$REPO_ROOT"/docs/ docs
+    for f in *.sol
+    do
+        echo "trying $f"
+        set +e
+        output=$("$SOLC" "$f" 2>&1)
```

Comment:
This is a copy of the previous test block. @chriseth if you could transform it into a bash function that could simplify it.